libraries/lib-project-file-io/SqliteSampleBlock.cpp

```diff
@@ -289,6 +298,31 @@ SampleBlockPtr SqliteSampleBlockFactory::DoCreateFromXML(
    return sb;
 }
 
+BlockSampleView SqliteSampleBlock::GetFloatSampleView()
+{
+   assert(mSampleCount > 0);
+
+   // First try to acquire the cache without locking. If that doesn't succeed,
+   // lock and try again. Can't remember how this pattern is called.
+   auto cache = mCache.lock();
+   if (cache)
+      return cache;
+
+   std::lock_guard<std::mutex> lock(mCacheMutex);
+   cache = mCache.lock();
+   if (cache)
+      return cache;
+
+   const auto newCache =
+      std::make_shared<std::vector<float>>(mSampleCount);
+   const auto cachedSize = DoGetSamples(
+      reinterpret_cast<samplePtr>(newCache->data()), floatSample, 0,
+      mSampleCount);
```

Comment:
Notice this `mSampleCount` correctly does NOT have type `sampleCount` but instead `size_t` because it is for an in-memory block.